pkg/persistence/config/writer/config_writer_test.go

```diff
@@ -1023,6 +1026,47 @@ func TestWriteConfigs(t *testing.T) {
 				"project/bucket/mybucket.json",
 			},
 		},
+		{
+			name:    "Grail file-segment",
+			envVars: map[string]string{featureflags.Temporary[featureflags.GrailFilterSegment].EnvName(): "true"},
+			configs: []config.Config{
```

Comment:
I'd also add here a test with the FF off that an error occurs

Reply:
Good suggestion.